docs/data/data-grid/localization/localization.md

```diff
@@ -6,10 +6,8 @@ The default locale of MUI is English (United States). If you want to use other l
 
 ## Translation keys
 
-<!-- #default-branch-switch -->
-
 You can use the `localeText` prop to pass in your own text and translations.
-You can find all the translation keys supported in [the source](https://github.com/mui/mui-x/blob/HEAD/packages/grid/x-data-grid/src/constants/localeTextConstants.ts)
+You can find all the translation keys supported in [the source](https://github.com/mui/mui-x/blob/-/packages/grid/x-data-grid/src/constants/localeTextConstants.ts)
```

Comment:
A trick from https://github.com/mui/material-ui/pull/36729.

Reply:
Didn't we initially have it like this, but faced some issues? 🤔 
I can't remember now, but maybe @cherniavskii remembers? 🤔 🤷

Reply:
The last PR that touched this line is #4080. Which surfaces the value of the change (we had to update the link, so the older version of the docs is now broken). 

It works as expected in https://deploy-preview-11521--material-ui-x.netlify.app/x/react-data-grid/localization/#translation-keys.

Reply:
I will take my chances. If we find problem, let's reference https://github.com/mui/material-ui/pull/36729 once we do, this way, we can look at that PR links and easily find this back:

![Screenshot 2023-12-28 at 17 24 49](https://github.com/mui/mui-x/assets/3165635/bdae62ad-ed45-4a86-a241-288e0b7a3da7)

Reply:
Sure, let's try it. 👍 
But another nitpick for @mui/xgrid, WDYT, would it make sense to have this link branch dependent in case the file has differences between master and next? 🤔 
And what about legacy versions, like v5? When that becomes legacy, the link would still point to the newest current branch, which not be what we would expect. 🤷 
It could make sense to have a similar process for such links as we have with `<!-- #default-branch-switch -->`. 🙈

Reply:
@LukasTy I don't follow you. With this PR Netlify preview, the link points to https://github.com/mui/mui-x/blob/v7.0.0-alpha.6/packages/grid/x-data-grid/src/constants/localeTextConstants.ts

Reply:
Yes, everything looks great on next, but I was thinking about the next major and how we should/could handle links to source code in general. 😉